internal/broker/broker.go

```diff
@@ -393,14 +396,41 @@ func (b *Broker) IsAuthenticated(sessionID, authenticationData string) (string,
 	if err = b.updateSession(sessionID, session); err != nil {
 		return AuthDenied, "", err
 	}
-	return access, data, nil
+
+	data, err := result.toJSON()
+	if err != nil {
+		slog.Debug(fmt.Sprintf("Impossible to convert to JSON %#v: %v", result, err))
+		if result.access == AuthDenied {
+			return result.access, data, nil
+		}
+		return AuthRetry, `{"message": "authentication data encoding failure"}`, nil
+	}
+
+	return result.access, data, nil
+}
+
+type isAuthenticatedResult struct {
+	access   string          `json:"-"`
+	UserInfo json.RawMessage `json:"userinfo,omitempty"`
+	Message  string          `json:"message,omitempty"`
 }
 
-func (b *Broker) handleIsAuthenticated(ctx context.Context, session *sessionInfo, authData map[string]string) (access, data string) {
+func (iar *isAuthenticatedResult) toJSON() (string, error) {
```

Comment:
You should implement `MarshalJSON` interface rather so that you don’t have a call an annex function when you want to marshall and this is more idiomatic.